include/ddc/create_mirror.hpp

```diff
@@ -13,118 +13,117 @@
 
 namespace ddc {
 
-/// Returns a new `Chunk` with the same layout as `src` allocated on the memory space `Space::memory_space`.
+/// @param[in] space A Kokkos memory space or execution space.
+/// @param[in] src A layout right ChunkSpan.
+/// @return a `Chunk` with the same support and layout as `src` allocated on the `Space::memory_space` memory space.
 template <class Space, class ElementType, class Support, class Layout, class MemorySpace>
 auto create_mirror(
         [[maybe_unused]] Space const& space,
         ChunkSpan<ElementType, Support, Layout, MemorySpace> const& src)
 {
+    static_assert(
+            Kokkos::is_memory_space_v<Space> || Kokkos::is_execution_space_v<Space>,
+            "DDC: parameter \"Space\" must be either a Kokkos execution space or a memory space");
     static_assert(
             std::is_same_v<Layout, std::experimental::layout_right>,
-            "Only layout right is supported");
+            "DDC: parameter \"Layout\" must be a `layout_right`");
     return Chunk(
             src.domain(),
             KokkosAllocator<std::remove_const_t<ElementType>, typename Space::memory_space>());
 }
 
-/// Returns a new host `Chunk` with the same layout as `src`.
+/// Equivalent to `create_mirror(Kokkos::HostSpace(), src)`.
+/// @param[in] src A layout right ChunkSpan.
+/// @return a `Chunk` with the same support and layout as `src` allocated on the `Kokkos::HostSpace` memory space.
 template <class ElementType, class Support, class Layout, class MemorySpace>
 auto create_mirror(ChunkSpan<ElementType, Support, Layout, MemorySpace> const& src)
 {
-    static_assert(
-            std::is_same_v<Layout, std::experimental::layout_right>,
-            "Only layout right is supported");
-    return Chunk(src.domain(), HostAllocator<std::remove_const_t<ElementType>>());
+    return create_mirror(Kokkos::HostSpace(), src);
 }
 
-/// Returns a new `Chunk` with the same layout as `src` allocated on the memory space `Space::memory_space` and operates a deep copy between the two.
+/// @param[in] space A Kokkos memory space or execution space.
+/// @param[in] src A layout right ChunkSpan.
+/// @return a `Chunk` with the same support and layout as `src` allocated on the `Space::memory_space` memory space and operates a deep copy between the two.
 template <class Space, class ElementType, class Support, class Layout, class MemorySpace>
 auto create_mirror_and_copy(
         Space const& space,
         ChunkSpan<ElementType, Support, Layout, MemorySpace> const& src)
 {
-    static_assert(
-            std::is_same_v<Layout, std::experimental::layout_right>,
-            "Only layout right is supported");
```

Comment:
Why is this static assert removed?

Reply:
As this function calls `create_mirror` which already does the check I removed it. I don't have yet a clear rule when to repeat the `static_assert` or not. Maybe I should reintroduce the check here because the function is not simple enough to remove it ?

Reply:
Up to you if it exists elsewhere then this should be fine